src/D2L.Bmx/UpdateHandler.cs

```diff
@@ -0,0 +1,136 @@
+using System.IO.Compression;
+using System.Runtime.InteropServices;
+using System.Formats.Tar;
+using System.Reflection;
+
+namespace D2L.Bmx;
+
+internal class UpdateHandler {
+
+	public async Task HandleAsync() {
+		using var httpClient = new HttpClient();
+		var releaseData = await UpdateChecker.GetLatestReleaseDataAsync();
+		var localVersion = Assembly.GetExecutingAssembly().GetName().Version;
+		var latestVersion = new Version( UpdateChecker.GetLatestReleaseVersion( releaseData ) );
+		if ( latestVersion <= localVersion ) {
+			Console.WriteLine( $"Already own the latest version {latestVersion}" );
+			return;
+		}
+		string archiveName = GetOSFileName();
+		string downloadUrl = releaseData?.Assets?.FirstOrDefault( a => a.Name == archiveName )?.BrowserDownloadUrl
+			?? string.Empty;
+		if( string.IsNullOrWhiteSpace( downloadUrl ) ) {
+			return;
```

Comment:
returning means we're cancelling the update and the user won't know what happened if we just kill the run